[TASK] Add pagetype icons

davidsteeb opened the pull request:
Related: BEXT-227

davidsteeb committed:
[TASK] Add pagetype icons

Related: BEXT-227

jschlier commented on Configuration/TCA/Overrides/pages.php:
```diff
@@ -72,6 +75,6 @@
         '--palette--;LLL:EXT:newspage/Resources/Private/Language/locallang_be.xlf:news;tx_newspage,',
         $dokType,
-        'after:title'
+        'after:title',
```
This comma shouldn't be here.
I don't like trailing commas, especially on function calls that have a set amount of parameters that will not change.

Additionally this breaks with PHP < 7.3. This is a conflict with TYPO3 10 PHP requirements (>= 7.2, <= 7.4) which I do not want to introduce.

jschlier commented on Configuration/TCA/Overrides/pages.php:
```diff
@@ -8,6 +8,9 @@
     // adding the new doktypes to the type select
     \TYPO3\CMS\Core\Utility\ExtensionManagementUtility::addTcaSelectItem('pages', 'doktype', $newsType);
 
+    $GLOBALS['TCA']['pages']['ctrl']['typsicon_classes'][$dokType] = 'apps-pagetree-newspage-page';
```
This should say `typeicon_classes` not `typsicon_classes`

jschlier commented on ext_localconf.php:
```diff
@@ -47,3 +47,27 @@
+    'apps-pagetree-newspage-tag-hideinmenu' => 'apps-pagetree-newspage-tag-hideinmenu',
+    'mimetypes-newspage' => 'mimetypes-newspage'
+];
+$iconRegistry = \TYPO3\CMS\Core\Utility\GeneralUtility::makeInstance(\TYPO3\CMS\Core\Imaging\IconRegistry::class);
```
When declaring variables in any of the configuration files, I prefer them to be enclosed in a function block.
This makes sure that they are not kept in memory and are "available" in other PHP files.

jschlier reviewed (changes requested):
A couple of small things should be adjusted.
I'll quickly add those changes myself 👍

jschlier committed:
[TASK] Remove trailing comma to avoid PHP version requirement conflict

Trailing commas in method calls are allowed with PHP versions >7.3. Adding this requirement would conflict TYPO3 10 PHP requirements, which requires PHP 7.2-7.4.

jschlier committed:
[BUGFIX] Fix typo when registering page type icons

jschlier committed:
[TASK] Set variables and configuration in function block

This makes sure that variables used are only set while required.
